lib/download.gi

```diff
@@ -0,0 +1,219 @@
+##############################################################################
+##
+#W  download.gi                 GAP4 package `Utils'             Thomas Breuer
+##
+#Y  Copyright (C) 2022, The GAP Group
+
+#############################################################################
+##
+#V  Download_Methods
+##
+##  Use the following tools (in this order).
+##
+##  - If the curlInterface package is available then call its
+##    'DownloadURL' function.
+##  - If the URL starts with 'http://' and if the IO package is available
+##    then call the 'SingleHTTPRequest' function from this package.
+##  - If a 'wget' executable is available then call it.
+##  - If a 'curl' executable is available then call it.
+##
+##  Note that currently the methods are *NOT* consistent in the case of
+##  failures:
+##
+##  - The function 'SingleHTTPRequest' does not regard the error code 301
+##    as failure.
```

Comment:
Actually, I wouldn't want it to treat it as a failure; I'd rather have it follow
the redirect... Do the other methods treat this as a failure? At least my `wget` says it follows up to 20 redirects by default. Aha, for `curl` one should set the `-L` resp. `--location` option to match what `wget` does -- I think this is a useful thing to do by default? Aha, and I see, we do give this option to `curl`. And `curlInterface` sets `followRedirect` to true by default, too.

So perhaps the text should say:

```suggestion
##  - The function 'SingleHTTPRequest' does not follow redirects as indicated
##    by HTTP status codes 301 and 302
```